server/src/main/java/io/crate/metadata/pgcatalog/PgTypeTable.java

```diff
@@ -64,31 +62,10 @@
             .add("typtypmod", INTEGER, c -> -1)
             .add("typnamespace", INTEGER, c -> TYPE_NAMESPACE_OID)
             .add("typarray", INTEGER, PGType::typArray)
-            .add("typinput", REGPROC, t -> {
-                if (t.typArray() == 0) {
-                    return Regproc.of("array_in");
-                } else {
-                    return regprocForMetaFunction(t, "_in");
-                }
-            })
-            .add("typoutput", REGPROC, t -> {
-                if (t.typArray() == 0) {
-                    return Regproc.of("array_out");
-                } else {
-                    return regprocForMetaFunction(t, "_out");
-                }
-            })
-            .add("typreceive", REGPROC, t -> regprocForMetaFunction(t, "recv"))
+            .add("typinput", REGPROC, PGType::typInput)
+            .add("typoutput", REGPROC, PGType::typOutput)
+            .add("typreceive", REGPROC, PGType::typReceive)
             .add("typnotnull", BOOLEAN, c -> false)
             .build();
     }
-
-    private static Regproc regprocForMetaFunction(PGType<?> type,
-                                                  String suffix) {
-        if (PGTypes.fromOID(type.oid()) != null) {
```

Comment:
Are you sure about all the types? I've added this check coz some types, like `any` (maybe it is actually the only type) don't have `typreceive`,  `typoutput`, and `typinput`

```
template1=# select typreceive from pg_type where oid = 2276;
 typreceive
------------
 -
(1 row)
```

Reply:
Good point. I added a fixup.